transforms/tabular/tabular-merger-plugin/src/polus/plugins/transforms/tabular/tabular_merger/tabular_merger.py

```diff
@@ -0,0 +1,289 @@
+    occurrences = {k: v for k, v in Counter(my_dict.values()).items()}
+
+    for k, v in my_dict.items():
+        count = occurrences[v]
+        if count > 1:
+            increment = v + 1
+            my_dict[k] = increment
+            occurrences[v] = count - 1
+
+    sorted_values = sorted(my_dict.values(), reverse=True)
+    i = 0
+    status = "Unknown"
+    prf = []
+    while status != "true":
+        for o in sorted_values:
+            for k, v in my_dict.items():
+                if v == o:
+                    status = "true"
+                    prf.append(k)
+                    i += 1
+
+    return prf
+
+
+def remove_files(curr_dir: pathlib.Path) -> None:
+    """Delete intermediate hdf5 and yaml files in a working directory.
+
+    Args:
+        curr_dir: Path to the working directory.
+    """
+    for f in curr_dir.iterdir():
+        if f.suffix in [".hdf5", ".yaml"]:
+            os.remove(f)
+
+
+def merge_files(
+    inp_dir_files: List,
+    strip_extension: bool,
+    file_extension: Union[Extension, str],
+    dim: Union[Dimensions, str],
+    same_rows: Optional[bool],
+    same_columns: Optional[bool],
+    map_var: Optional[str],
+    out_dir: str,
+) -> None:
+    """Merge tabular files with vaex supported file formats into a single combined file using either row or column merging.The merge file can be save into any of the vaex supported file format.
+
+    Args:
+        inp_dir_files: List of an input files.
+        file_pattern : Pattern to parse input files.
+        strip_extension:  True to remove csv from the filename in the output file.
+        file_extension: File format of an output merged file
+        dim: To perform merging either `rows` or `columns` wise
+        same_rows:  Only merge csv files with the same number of rows.
+        same_columns: Check for common header and then perform merging of files with common column names.
+        map_var: Variable Name used to join file column wise.
+        out_dir:Path to output directory
+    """
+    if isinstance(file_extension, str):
+        file_extension = Extension(file_extension)
+    if isinstance(dim, str):
+        dim = Dimensions(dim)
+
+    if file_extension == Extension.Default:
+        file_extension = POLUS_TAB_EXT
+    elif file_extension == Extension.CSV:
+        file_extension = Extension.CSV.value
+    elif file_extension == Extension.ARROW:
+        file_extension = Extension.ARROW.value
+    elif file_extension == Extension.PARQUET:
+        file_extension = Extension.PARQUET.value
+    elif file_extension == Extension.HDF:
+        file_extension = Extension.HDF.value
+    elif file_extension == Extension.FEATHER:
+        file_extension = Extension.FEATHER.value
+
+    if dim == Dimensions.Default:
+        dim = Dimensions.Rows.value
+    elif dim == Dimensions.Rows:
+        dim = Dimensions.Rows.value
+    elif dim == Dimensions.Columns:
+        dim = Dimensions.Columns.value
```

Comment:
These are not necessary. The type should guarantee that this always succeeds.

Reply:
done